src/node/internal/internal_tls_wrap.ts

```diff
@@ -0,0 +1,644 @@
+      isServer?: boolean;
+      requestOCSP?: boolean;
+      server?: unknown;
+      onread?: OnReadOpts;
+    };
+  public _secureEstablished: boolean;
+  public _securePending: boolean;
+  public _newSessionPending: boolean;
+  public _controlReleased: boolean;
+  public _parent: Socket;
+  public authorized: boolean;
+  public encrypted: boolean;
+  public handle: ReturnType<TLSSocket['_wrapHandle']>;
+  public servername: null | string;
+  public secureConnecting: boolean;
+  public ssl: TLSSocket['_handle'];
+  public [kRes]: null | Socket['_handle'];
+  public [kIsVerified]: boolean;
+  public [kPendingSession]: null | Buffer;
+  public [kErrorEmitted]: boolean;
+  public [kConnectOptions]?: NormalizedConnectionOptions;
+
+  public constructor(
+    socket: Socket | Duplex | undefined,
+    opts: TLSSocket['_tlsOptions']
+  );
+  public prototype: TLSSocket;
+
+  public _destroySSL(): void;
+  public _emitTLSError(error: Error): void;
+  public _finishInit(): void;
+  public _handleTimeout(): void;
+  public _releaseControl(): boolean;
+  public _start(): void;
+  public _tlsError(error: Error): Error | null;
+  public _wrapHandle(
+    wrap: null | Socket,
+    handle: Socket['_handle'] | null | undefined,
+    wrapHasActiveWriteFromPrevOwner: boolean
+  ): unknown;
+  public disableRenegotiation(): void;
+  public getX509Certificate(): ReturnType<TLSSocketType['getX509Certificate']>;
+  public setKeyCert(context: unknown): void;
+  public setServername(name: string): void;
+  public setSession(session: string | Buffer): void;
+  public setMaxSendFragment(size: number): boolean;
+  public getCertificate(): ReturnType<TLSSocketType['getCertificate']>;
+  public getPeerX509Certificate(): ReturnType<
+    TLSSocketType['getPeerX509Certificate']
+  >;
+  public renegotiate(
+    options: {
+      rejectUnauthorized?: boolean | undefined;
+      requestCert?: boolean | undefined;
+    },
+    callback?: (error: Error | null) => void
+  ): boolean;
+  public exportKeyingMaterial(
+    length: number,
+    label: string,
+    context?: Buffer
+  ): Buffer;
+}
+
+function onnewsessionclient(
+  this: TLSSocket,
+  _sessionId: string,
+  session: Buffer
+): void {
+  if (this[kIsVerified]) {
+    this.emit('session', session);
+  } else {
+    this[kPendingSession] = session;
+  }
+}
+
+function onerror(this: TLSSocket, err: Error): void {
+  if (this._hadError) return;
+
+  this._hadError = true;
+
+  // Destroy socket if error happened before handshake's finish
+  if (!this._secureEstablished) {
+    // When handshake fails control is not yet released,
+    // so self._tlsError will return null instead of actual error
+
+    // Set closing the socket after emitting an event since the socket needs to
+    // be accessible when the `tlsClientError` event is emitted.
+    this.destroy(err);
+  } else {
+    // Emit error
+    this._emitTLSError(err);
+  }
+}
+
+// @ts-expect-error TS2323 Cannot redeclare error.
+export function TLSSocket(
```

Comment:
Add a small comment explaining that we're using old style function classes for node.js compat